src/applications/public-outreach-materials/libraries/library-filters.js

```diff
@@ -55,9 +70,47 @@ export function libraryCurrent() {
   });
 }
 
+export function libraryPagerGen() {
+  numCards = libraryNumCards();
+
+  if (document.getElementById('pager-nums-insert')) {
+    const diff = pages - activePage;
+    let pagerHtml;
+
+    pagerHtml = `<button id="va-pagination-active-num" class="va-page-numbers">
+    ${activePage}</button>`;
+    if (diff > 1 && (numCards === undefined || numCards > 9)) {
+      pagerHtml += `<a class="pager-numbers" aria-label="Load page
+      ${activePage + 1}" tabindex="${activePage + 1}">${activePage + 1}</a>`;
+    }
+    if (diff > 2 && (numCards === undefined || numCards > 9)) {
+      pagerHtml += `<a class="pager-numbers"" aria-label="Load page
+      ${activePage + 2}" tabindex="${activePage + 2}">${activePage + 2}</a>`;
+    }
+    if (diff > 3 && (numCards === undefined || numCards > 9)) {
+      pagerHtml += `.... <a class="pager-numbers"" aria-label="Load page
+      ${pages}" tabindex="${pages}"> ${pages}</a>`;
+    }
+    document.getElementById('pager-nums-insert').innerHTML = pagerHtml;
+  }
+}
+
+export function libraryReset() {
+  libraryPagerGen();
+  libraryCurrent();
+  libraryCount();
+}
+
 export function libraryFilters(el) {
```

Comment:
This function seems like it has a lot of responsibilities. Is it possible to break them up into sub-functions for clarity? Even comments that add context would go a long way for readability.

Reply:
Good point. I'll add some comments, @cvalarida